build_integration_test.sh

```diff
@@ -94,31 +94,83 @@ run_and_test() {
 
     wait_for_function_up $function_name $port $timeout
 
-    curl -s 127.0.0.1:$port > got.txt
+    got=$(curl -s 127.0.0.1:${port})
+    want="Function output from integration testing: Hello World!"
 
-cat << EOF > want.txt
-Function output from integration testing: Hello World!
-EOF
+    echo "Removing container..."
+    docker rm -f $id
 
-    if cmp got.txt want.txt ; then
+    echo "Removing function image $function_name:latest"
+    docker rmi -f $function_name:latest
+
+    echo "Removing created files..."
+    rm -rf $function_name*
+
+    if [ "$got" == "$want" ]; then
         echo "SUCCESS testing function $function_name"
     else
         echo "FAILED testing function $function_name"
+        exit 1
     fi
+}
 
-    echo "Removing container..."
-    docker rm -f $id
+build_stack_when_yaml_is_invalid() {
```

Comment:
Hi

These tests were not meant to be expanded with additional test-cases around the CLI. They are present to check the templating feature only. 

I think if this needs testing it should be stubbed/mocked out properly in the Golang integration tests.

Reply:
I agree, mocking out the docker commands in Golang would be better than the integration tests.